runtime/devnet/src/config/contracts.rs

```diff
@@ -44,6 +45,11 @@ parameter_types! {
 	pub const CodeHashLockupDepositPercent: Perbill = Perbill::from_percent(0);
 }
 
+impl pop_chain_extension::Config for Runtime {
```

Comment:
As mentioned elsewhere, while I like the idea I think the implementation is a little confusing as we have all these dependent traits littered about which may make maintenance more difficult.

A rough suggestion might be to have an Extension or ChainExtension type defined and the relevant traits required implemented for that type only, so it then just effectively 'plugs in' to the ChainExtension on line 72.

I'll try find some time later today to try and mock something out.

Reply:
@evilrobot-01 Thank you for a very thoughtful feedback. I created a sub-PR: https://github.com/r0gue-io/pop-node/pull/176 to show the alternative approach which removes the `Config` trait so on the Runtime side, it will be cleaner when there is no involvement between the `Runtime` and the extension configuration. However, this approach makes the extension crate code a bit more complex as there are generic types passing around between functions. Could you please skim over it and let me know if this is the implementation that we expect?